src/java/org/apache/cassandra/io/util/DataOutputPlus.java

```diff
@@ -129,4 +129,41 @@ default boolean hasPosition()
     {
         return false;
     }
+
+    // The methods below support page-aware layout for writing. These would only be implemented if position() is
+    // also supported.
+
+    /**
+     * Returns the number of bytes that a page can take at maximum.
+     */
+    default int maxBytesInPage()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Pad this page with 0s to move on to the next.
+     * @throws IOException
```

Comment:
nit: Don't need this if it's already in the `throws` clause unless we have more detailed info